component/module/git/internal/vcs/git.go

```diff
@@ -103,15 +103,16 @@ func isRepoCloned(dir string) bool {
 // Update updates the repository by fetching new content and re-checking out to
 // latest version of Revision.
 func (repo *GitRepo) Update(ctx context.Context) error {
-	err := repo.repo.FetchContext(ctx, &git.FetchOptions{
+	var err error
+	fetchRepoErr := repo.repo.FetchContext(ctx, &git.FetchOptions{
 		RemoteName: "origin",
 		Force:      true,
 		Auth:       repo.opts.Auth.Convert(),
 	})
-	if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
-		return UpdateFailedError{
+	if fetchRepoErr != nil && !errors.Is(fetchRepoErr, git.NoErrAlreadyUpToDate) {
+		err = UpdateFailedError{
 			Repository: repo.opts.Repository,
-			Inner:      err,
+			Inner:      fetchRepoErr,
```

Comment:
Ditto here: I'm not sure we should be falling back to resetting the git repo when the update fails, and I think we should just return directly here. 

(Also, this instance of `err` is always overwritten by the assignment on line 125 below)

Reply:
Good point here. I've amended in latest commit.